src/blockdigest.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef _BITCOIN_SRC_BLOCKDIGEST_H_
+#define _BITCOIN_SRC_BLOCKDIGEST_H_
+
+#include <vector>
+#include <boost/filesystem.hpp>
+
+#include "bloom.h"
+#include "serialize.h"
+#include "uint256.h"
+
+class COutPoint;
+class CBlock;
+class CTransaction;
+
+namespace DigestFilter {
+
+enum format {
+    vectorized      = 1 << 0,   // argument is a uint8_t indicating size
+    mapped          = 1 << 1,   // argument comprises two uint8_t's indicating offset and count
+    inverted        = 1 << 2,   // listed elements are ABSENT; unlisted elements are PRESENT
+};
+
+typedef std::vector<uint8_t> data_t;
+
+class base_filter {
+public:
+    virtual void insert(const data_t& data) = 0;
+    virtual void clear() = 0;
+    virtual bool contains(const data_t& data) const = 0;
+    
+    data_t outpoint_data(const COutPoint& outpoint) const;
+    data_t hash_data(const uint256& hash) const;
+    std::vector<const data_t> tx_data_vec(const CTransaction& tx) const;
+    
+    void digest_block(const CBlock& block);
+    void digest_tx(const CTransaction& tx);
+
+    virtual void insert(const std::vector<const data_t>& data_vec) {
+        for (const data_t& d : data_vec) insert(d);
+    }
+
+    virtual bool contains(const std::vector<const data_t>& data_vec) const {
+        for (const data_t& d : data_vec) if (contains(d)) return true;
+        return false;
+    }
+
+    virtual void insert(const COutPoint& outpoint) { insert(outpoint_data(outpoint)); }
+    virtual void insert(const uint256& hash)       { insert(hash_data(hash)); }
+    virtual void insert(const CTransaction& tx)    { insert(tx_data_vec(tx)); }
+
+    virtual bool contains(const COutPoint& outpoint) const { return contains(outpoint_data(outpoint)); }
+    virtual bool contains(const uint256& hash) const       { return contains(hash_data(hash)); }
+    virtual bool contains(const CTransaction& tx) const    { return contains(tx_data_vec(tx)); }
+};
+
+class bloom_filter : public base_filter {
+private:
+    CBloomFilter* filter;
```

Comment:
Yeah, your copy constructor should be deleted :)

Also it's not quite clear to me why this even needs to be a pointer rather than a member, but maybe I'm missing something.